compiler/src/Language/Bond/Codegen/Java/Class_java.hs

```diff
@@ -373,6 +373,41 @@ public #{declName}(com.microsoft.bond.StructBondType<#{declName}<#{paramList}>>
             else mempty
 
 
+-- builds text for Object.equals(Object) override
+object_equals :: String -> [Field] -> Text
+object_equals declName fields = [lt|@Override
+    public boolean equals(Object o) {
+        if (o == null) return false;
+        if (!(o instanceof #{declName})) return false;
+        final #{declName} other = (#{declName}) o;
+        #{newlineSep 2 compareField fields}
+        return true;
+    }
+|]
+    where
+        compareField Field {..} = [lt|if (!(#{equals_member fieldType fieldName})) return false;|]
+        equals_member t f
+            | isJavaPrimitive t = [lt|this.#{f} == other.#{f}|]
+            | otherwise         = [lt|(this.#{f} == null && other.#{f} == null) || (this.#{f} != null && this.#{f}.equals(other.#{f}))|]
+
+
+-- builds text for Object.hashCode() override
+object_hashCode :: [Field] -> Text
+object_hashCode fields = [lt|@Override
+    public int hashCode() {
+        int result = 17;
+        #{newlineSep 2 hashField fields}
+        return result;
+    }
+|]
+    where
+        hashField Field {..} = [lt|result = (31 * result) + #{hash_member fieldType fieldName};|]
+        hash_member BT_Bool f   = [lt|(#{f} ? 0 : 1)|]
+        hash_member t f
+            | isJavaPrimitive t = [lt|(int) #{f}|]
```

Comment:
Casting a float/double to an int for hashing purposes doesn't seem like it will give good distribution, particularly if, say, the float is representing numbers in \[0, 1).